src/bentoml/_internal/frameworks/flax.py

```diff
@@ -0,0 +1,324 @@
+           )
+
+           logger.info(
+               "epoch:% 3d, train_loss: %.4f, train_accuracy: %.2f, test_loss: %.4f, test_accuracy: %.2f"
+               % (epoch, train_loss, train_accuracy * 100, test_loss, test_accuracy * 100)
+           )
+
+       # `Save` the model with BentoML
+       tag = bentoml.flax.save_model("mnist", CNN(), state)
+    """
+    if not isinstance(module, nn.Module):
+        raise BentoMLException(
+            f"'bentoml.flax.save_model()' only support saving 'flax.linen.Module' object. Got {module.__class__.__name__} instead."
+        )
+
+    context = ModelContext(
+        framework_name="flax",
+        framework_versions={
+            "flax": get_pkg_version("flax"),
+            "jax": get_pkg_version("jax"),
+            "jaxlib": get_pkg_version("jaxlib"),
+        },
+    )
+
+    if signatures is None:
+        signatures = {"__call__": {"batchable": False}}
+        logger.info(
+            'Using the default model signature for Flax (%s) for model "%s".',
+            signatures,
+            name,
+        )
+    custom_objects = {} if custom_objects is None else custom_objects
+    custom_objects["_module"] = module
+
+    with bentoml.models.create(
+        name,
+        module=MODULE_NAME,
+        api_version=API_VERSION,
+        signatures=signatures,
+        labels=labels,
+        options=FlaxOptions(),
+        custom_objects=custom_objects,
+        external_modules=external_modules,
+        metadata=metadata,
+        context=context,
+    ) as bento_model:
+        with open(bento_model.path_of(MODEL_FILENAME), "wb") as f:
+            f.write(serialization.to_bytes(state))
+        return bento_model
+
+
+def get_runnable(bento_model: bentoml.Model) -> t.Type[bentoml.Runnable]:
+    """
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
+    """
+    partial_kwargs: dict[str, t.Any] = bento_model.info.options.partial_kwargs
+    device: str = bento_model.info.options.device
+
+    class FlaxRunnable(bentoml.Runnable):
+        SUPPORTED_RESOURCES = ("nvidia.com/gpu", "cpu", "tpu")
+        SUPPORTS_CPU_MULTI_THREADING = True
+
+        def __init__(self):
+            super().__init__()
+
+            self.model, self.state_dict = load_model(bento_model, device=device)
+            self.params = self.state_dict["params"]
+            self.methods_cache: t.Dict[str, t.Callable[..., t.Any]] = {}
+
+    def gen_run_method(self: FlaxRunnable, method_name: str):
+        method = getattr(self.model, method_name)
+        method_partial_kwargs = partial_kwargs.get(method_name)
+        if method_partial_kwargs is not None:
+            method = functools.partial(method, **method_partial_kwargs)
+
+        def mapping(item: jnp.ndarray) -> jnp.ndarray:
+            if not LazyType["jnp.ndarray"]("jnp.ndarray").isinstance(item):
+                return jnp.asarray(item)
+            return item
+
+        def run_method(self: FlaxRunnable, *args: jnp.ndarray):
```

Comment:
so the type of  inputs of flax runner can be only `jnp.ndarray`? Should we consider adding support for ndarray? or other tensor format?

Reply:
I was thinking to only support jnp.ndarray for now. This is because we don't know how to process each data type, and users have to handle that for now.

Reply:
can we support `np.ndarray`?

Reply:
Yes.